src/clj/harja/palvelin/palvelut/yllapitokohteet/paikkauskohteet.clj

```diff
@@ -563,6 +562,191 @@
       (throw+ {:type "Error"
                :virheet [{:koodi "ERROR" :viesti "Ladatussa tiedostossa virhe."}]}))))
 
+(def paikkauksen-domain-avaimet
+  [::paikkaus/alkuaika
+   ::paikkaus/loppuaika
+   ::tr/tie
+   ::paikkaus/ajorata
+   ::tr/aosa
+   ::tr/aet
+   ::tr/losa
+   ::tr/let
+   ::paikkaus/leveys
+   ::paikkaus/ajourat
+   ::paikkaus/reunat
+   ::paikkaus/ajouravalit
+   ::paikkaus/keskisaumat
+   ::paikkaus/massatyyppi
+   ::paikkaus/raekoko
+   ::paikkaus/kuulamylly
+   ::paikkaus/massamaara
+   ::paikkaus/pinta-ala])
+
+(defn- tee-tr-osoite [paikkaus]
+  (-> paikkaus
+    (assoc ::paikkaus/tierekisteriosoite
+      (select-keys paikkaus [::tr/tie ::tr/aosa ::tr/aet ::tr/losa ::tr/let]))
+    (dissoc ::tr/tie ::tr/aosa ::tr/aet ::tr/losa ::tr/let)))
+
+(defn- tee-tienkohta [{::paikkaus/keys [ajourat ajorata ajouravalit reunat keskisaumat] :as paikkaus}]
+  (-> paikkaus
+    (assoc ::paikkaus/tienkohdat
+      (merge {::paikkaus/ajorata ajorata}
+        (when ajourat {::paikkaus/ajourat [ajourat]})
+        (when ajouravalit {::paikkaus/ajouravalit [ajouravalit]})
+        (when reunat {::paikkaus/reunat [reunat]})
+        (when keskisaumat {::paikkaus/keskisaumat [keskisaumat]})))
+    (dissoc ::paikkaus/ajourat ::paikkaus/ajorata ::paikkaus/ajouravalit ::paikkaus/reunat ::paikkaus/keskisaumat)))
+
+(defn- yrita-tehda-sijainti [{::tr/keys [tie aosa aet losa let] :as paikkaus} db]
+  (if (and tie aosa aet losa let)
+    (assoc paikkaus ::paikkaus/sijainti
+      (q-tr/tierekisteriosoite-viivaksi db {:tie tie :aosa aosa
+                                            :aet aet :losa losa
+                                            :loppuet let}))
+    paikkaus))
+
+(defn- tee-massamenekki [{pinta-ala ::paikkaus/pinta-ala
+                          massamaara ::paikkaus/massamaara :as paikkaus}]
+  (assoc paikkaus ::paikkaus/massamenekki
+    (when (and pinta-ala massamaara
+            (number? pinta-ala)
+            (number? massamaara))
+      ;; with-precision haluaa tarkkuuden, ei skaalaa.
+      ;; Koska ei voida tietää ennen desimaalia olevien määrää, joudutaan laskemaan tarkkuus hieman hankalasti.
+      (let [tarkkuus (+ 2 (count (str (int (/ (* 1000 (float massamaara))
+                                             (float pinta-ala))))))]
+        (with-precision
+          tarkkuus
+          (/ (* 1000 massamaara)
```

Comment:
Tää on sinänsä oikein, mutta jäin miettimään että pitäskö meidän koettaa saada vaikka paikkaus-domainiin yleinen funktio, malliin (defn massamaara-ja-pinta-ala->massamenekki [massamaara pinta-ala] ...
eli näitä muunnoksia alkaa nyt olemaan muutamassa paikassa, niin saataisiin ne keskitettyä yhteen ja samaan funktioon. Tuossa ylläolevakin muunnos (float massamäärä) voisi kenties mennä sillä samalla funktiolla, siellä sisällä voitaisiin ehkä tehdä tarvittavia tyyppitarkasteluja ja -muunnoksia?

Reply:
Hyvä idea, siirrän sinne.